proparse/src/main/java/org/prorefactor/core/ProToken.java

```diff
@@ -113,4 +115,28 @@ public int getEndFileIndex() {
     return endFile;
   }
 
+  @Override
+  public boolean equals(Object obj) {
+    if (obj instanceof ProToken) {
+      ProToken tok = (ProToken) obj;
+      return ((tok.type == this.type) && (tok.text.equals(this.text)) && (tok.line == this.line)
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Reduce the number of conditional operators (8) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1067)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove those useless parentheses. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AUselessParenthesesCheck)